packages/hardhat-ledger/src/provider.ts

```diff
@@ -0,0 +1,437 @@
+      });
+    }
+
+    if (txRequest.from === undefined) {
+      throw new HardhatError(ERRORS.NETWORK.MISSING_TX_PARAM_TO_SIGN_LOCALLY, {
+        param: "from",
+      });
+    }
+
+    const hasGasPrice = txRequest.gasPrice !== undefined;
+    const hasEip1559Fields =
+      txRequest.maxFeePerGas !== undefined ||
+      txRequest.maxPriorityFeePerGas !== undefined;
+
+    if (!hasGasPrice && !hasEip1559Fields) {
+      throw new HardhatError(ERRORS.NETWORK.MISSING_FEE_PRICE_FIELDS);
+    }
+
+    if (hasGasPrice && hasEip1559Fields) {
+      throw new HardhatError(ERRORS.NETWORK.INCOMPATIBLE_FEE_PRICE_FIELDS);
+    }
+
+    if (hasEip1559Fields && txRequest.maxFeePerGas === undefined) {
+      throw new HardhatError(ERRORS.NETWORK.MISSING_TX_PARAM_TO_SIGN_LOCALLY, {
+        param: "maxFeePerGas",
+      });
+    }
+
+    if (hasEip1559Fields && txRequest.maxPriorityFeePerGas === undefined) {
+      throw new HardhatError(ERRORS.NETWORK.MISSING_TX_PARAM_TO_SIGN_LOCALLY, {
+        param: "maxPriorityFeePerGas",
+      });
+    }
+
+    // If we don't manage the address, the method is forwarded
+    if (this._isControlledAddress(txRequest.from)) {
+      const path = await this._derivePath(txRequest.from);
+
+      if (txRequest.nonce === undefined) {
+        txRequest.nonce = await this._getNonce(txRequest.from);
+      }
+
+      const chainId = await this._getChainId();
+
+      const baseTx: ethers.utils.UnsignedTransaction = {
+        chainId,
+        data: txRequest.data,
+        gasLimit: txRequest.gas,
+        gasPrice: txRequest.gasPrice,
+        nonce: Number(txRequest.nonce),
+        value: txRequest.value,
+      };
+      if (txRequest.to !== undefined) {
+        baseTx.to = this._toHex(txRequest.to);
+      }
+
+      const txToSign = ethers.utils.serializeTransaction(baseTx).substring(2);
+
+      const resolution = await ledgerService.resolveTransaction(
+        txToSign,
+        {},
+        {}
+      );
+
+      const signature = await this._withConfirmation(() =>
+        this.eth.signTransaction(path, txToSign, resolution)
+      );
+
+      const rawTransaction = ethers.utils.serializeTransaction(baseTx, {
+        v: ethers.BigNumber.from(this._toHex(signature.v)).toNumber(),
+        r: this._toHex(signature.r),
+        s: this._toHex(signature.s),
+      });
+
+      return this._wrappedProvider.request({
+        method: "eth_sendRawTransaction",
+        params: [rawTransaction],
+      });
+    }
+  }
+
+  private async _derivePath(addressToFindAsBuffer: Buffer): Promise<string> {
+    const addressToFind = this._toHex(addressToFindAsBuffer).toLowerCase();
+
+    if (this.paths[addressToFind] !== undefined) {
+      return this.paths[addressToFind];
+    }
+
+    this.emit("derivation_start");
+
+    let path = "";
+    try {
+      for (
+        let account = 0;
+        account <= LedgerProvider.MAX_DERIVATION_ACCOUNTS;
+        account++
+      ) {
+        path = `44'/60'/${account}'/0'/0`;
```

Comment:
I think the last apostrophe is wrong? At least that's what BIP-44 says, and also I couldn't use an account in my ledger until I removed it.

If it is indeed wrong, we also need to update all the error messages that use explicit derivation paths.

Reply:
It is 🤦 , testing residue, great catch. Changed to `44'/60'/0/0/0`